File: problems/0206_reverse-linked-list/solution.ts
function reverseList(head: ListNode | null): ListNode | null {
    let prev: ListNode | null = null;
    let current: ListNode | null = head;
    while (current !== null) {
        const next: ListNode | null = current.next;
        current.next = prev;
        prev = current;
        current = next;
    }
    return prev;
}

File: problems/0206_reverse-linked-list/solution.cpp
class Solution {
  public:
    ListNode *reverseList(ListNode *head) {
        ListNode *prev = nullptr;
        ListNode *current = head;
        while (current != nullptr) {
            ListNode *next = current->next;
            current->next = prev;
            prev = current;
            current = next;
        }
        return prev;
    }
};
